File: Application/src/main/java/com/shareyourproxy/api/rx/JustSingle.java
package com.shareyourproxy.api.rx;

import android.util.Log;

import rx.SingleSubscriber;
import timber.log.Timber;

/**
 * Single subscriber that logs.
 */
public class JustSingle<T> extends SingleSubscriber<T> {
    @Override
    public void onSuccess(T value) {
        Timber.v("%1$s Success obj: %2$s", this.toString(), value.toString());
    }

    @Override
    public void onError(Throwable e) {
        Timber.e(Log.getStackTraceString(e));
    }
}

File: Application/src/main/kotlin/com/shareyourproxy/api/rx/JustSingle.kt
package com.shareyourproxy.api.rx

import android.util.Log

import rx.SingleSubscriber
import timber.log.Timber

/**
 * Single subscriber that logs.
 */
internal abstract class JustSingle<T>(private val clazz:Class<*>) : SingleSubscriber<T>() {
    override fun onSuccess(value: T) {
        Timber.v("${clazz.simpleName} Success obj: ${value.toString()}")
    }

    override fun onError(e: Throwable) {
        Timber.e("${clazz.simpleName}: ${Log.getStackTraceString(e)}")
    }
}
